components/style/matching.rs

```diff
@@ -635,6 +627,40 @@ pub trait MatchMethods : TElement {
         }
     }
 
+    /// Updates the style attribute rule nodes without re-running selector
+    /// matching, using just the rule tree.
+    fn update_style_attribute(&self,
+                              context: &StyleContext<Self>,
+                              data: &mut AtomicRefMut<ElementData>)
+                              -> MatchResults {
+        let style_attribute = self.style_attribute();
+
+        let mut new_rule_node = data.styles().primary.rules.clone();
```

Comment:
Nit: I'd just call this |rule_node|, because it's not new at this point, and it's confusing when we pass the old rule node to replace_rule_level_if_applicable as |new_rule_node|.